### PR TITLE
Removed all all-zero masks

### PR DESCRIPTION
**Pull request by kywch**

### CODE REVIEW

**Comment by jsuarez5341** [comment]
We learned some stuff from this:
- It does not appear that we have all 0 masks in the forward pass, yet somehow 0 entropy and very negative logits are leaking into the training batch
- KL is already exploding by the time entropy crashes

Probably, we are going to need to stain the masked items with large, unique negative values to figure out what is leaking. This is likely part of a bigger problem with the complexity introduced into clean_pufferl to handle variable numbers of agents
